app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    user.update(posts_counter: user.posts.count)
+  end
```

Comment:
Optional suggestion:

I would recommend using  [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter`. Something like that:

```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_count)
  end
```

Reply:
Thanks @Emmyn5600 for this tip. This was interesting. I didn't know this.

But I would say I disagree with using increment. What if we remove one post? counter only gets incremented.

I will use `after_save` but still I put the same method in it. Unless we have mechanism for reducing the counter, we shouldn't increment without taking the actual existing number of posts into the account.

Am I wrong?